File: app/src/main/java/com/jodelXposed/hooks/BetaStuff.java
package com.jodelXposed.hooks;

import com.jodelXposed.models.Hookvalues;
import com.jodelXposed.utils.Options;

import de.robv.android.xposed.XC_MethodHook;
import de.robv.android.xposed.callbacks.XC_LoadPackage;

import static de.robv.android.xposed.XposedHelpers.findAndHookMethod;

public class BetaStuff {

    public BetaStuff(XC_LoadPackage.LoadPackageParam lpparam) {
        Hookvalues hooks = Options.getInstance().getHooks();

        // Unlock experiments (features that are available on some devices like post pining or searching for hashtags)
        findAndHookMethod(hooks.Class_Storage, lpparam.classLoader, hooks.BetaHook_UnlockFeatures, String.class, new XC_MethodHook() {
            @Override
            protected void afterHookedMethod(MethodHookParam param) throws Throwable {
                param.setResult(true);
            }
        });

    }
}

File: app/src/main/java/com/jodelXposed/hooks/BetaStuff.kt
package com.jodelXposed.hooks

import com.jodelXposed.utils.Log
import com.jodelXposed.utils.Options
import de.robv.android.xposed.XC_MethodHook
import de.robv.android.xposed.callbacks.XC_LoadPackage.LoadPackageParam
import de.robv.android.xposed.XposedHelpers.findAndHookMethod

class BetaStuff(lpparam: LoadPackageParam) {
    init {
        // Unlock experiments (features that are available on some devices like post pining or searching for hashtags)
        findAndHookMethod(Options.hooks.Class_Storage, lpparam.classLoader, Options.hooks.Method_BetaHook_UnlockFeatures, String::class.java, object : XC_MethodHook() {
            @Throws(Throwable::class)
            override fun afterHookedMethod(param: MethodHookParam) {
                val requested: String = param.args[0] as String
                Log.vlog("Requested feature: $requested")
                Log.vlog("Default enabled: ${param.result}")
                for (feature in Options.hooks.Array_FeaturesEnabled) {
                    with(feature) {
                        if (startsWith("!") && substring(1) == requested) {
                            param.result = false
                            Log.vlog("Feature disabled")
                        } else if (param.result == false && equals(requested)) {
                            param.result = true
                            Log.vlog("Feature enabled")
                        }
                    }
                }
            }
        })
    }
}